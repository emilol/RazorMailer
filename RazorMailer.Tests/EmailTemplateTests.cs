using System.Threading.Tasks;
using Assent;
using RazorMailer.Models;
using RazorMailer.Parsing;
using Xunit;

namespace RazorMailer.Tests
{
    public class EmailTemplateTests
    {
        [Fact]
        public async Task HelloWorldModel()
        {
            // Arrange
            var template = "/Templates/HelloWorld.cshtml";
            var model = new HelloWorldModel
            {
                FirstName = "Joe",
                LastName = "Smith"
            };
            var templateParser = new TemplateParser();

            // Act
            var result = await templateParser.RenderAsync(template, model);

            // Assert
            this.Assent(result);
        }

        [Fact]
        public async Task NoModel()
        {
            // Arrange
            var template = "/Templates/NoModel.cshtml";
            var templateParser = new TemplateParser();

            // Act
            var result = await templateParser.RenderAsync(template);

            // Assert
            this.Assent(result);
        }
    }
}