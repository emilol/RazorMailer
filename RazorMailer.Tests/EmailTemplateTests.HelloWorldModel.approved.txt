<p>Hello Joe Smith!</p>